src/plugins/data_source_management/public/components/edit_data_source/edit_data_source.tsx

```diff
@@ -51,49 +51,48 @@ export const EditDataSource: React.FunctionComponent<RouteComponentProps<{ id: s
   /* Fetch data source by id*/
   useEffectOnce(() => {
     (async function () {
-      setIsLoading(true);
-      try {
-        const fetchDataSourceById = await getDataSourceById(dataSourceID, savedObjects.client);
-        const listOfDataSources: DataSourceTableItem[] = await getDataSources(savedObjects.client);
-        if (fetchDataSourceById) {
-          setDataSource(fetchDataSourceById);
-          setBreadcrumbs(getEditBreadcrumbs(fetchDataSourceById));
-        }
-        if (Array.isArray(listOfDataSources) && listOfDataSources.length) {
-          setExistingDatasourceNamesList(
-            listOfDataSources.map((datasource) => datasource.title?.toLowerCase())
-          );
-        }
-      } catch (e) {
-        setDataSource(defaultDataSource);
-        handleDisplayToastMessage({
-          id: 'dataSourcesManagement.editDataSource.editDataSourceFailMsg',
-          defaultMessage: 'Unable to find the Data Source.',
-        });
-        props.history.push('');
-      } finally {
-        setIsLoading(false);
-      }
+      await fetchDataSourceDetailsByID();
     })();
   });
 
-  /* Handle submit - create data source*/
-  const handleSubmit = async (attributes: DataSourceAttributes) => {
+  const fetchDataSourceDetailsByID = async () => {
     setIsLoading(true);
     try {
-      await updateDataSourceById(savedObjects.client, dataSourceID, attributes);
-      props.history.push('');
+      const fetchDataSourceById = await getDataSourceById(dataSourceID, savedObjects.client);
+      const listOfDataSources: DataSourceTableItem[] = await getDataSources(savedObjects.client);
+      if (fetchDataSourceById) {
+        setDataSource(fetchDataSourceById);
+        setBreadcrumbs(getEditBreadcrumbs(fetchDataSourceById));
+      }
+      if (Array.isArray(listOfDataSources) && listOfDataSources.length) {
+        setExistingDatasourceNamesList(
+          listOfDataSources.map((datasource) => datasource.title?.toLowerCase())
+        );
+      }
     } catch (e) {
-      setIsLoading(false);
+      setDataSource(defaultDataSource);
       handleDisplayToastMessage({
         id: 'dataSourcesManagement.editDataSource.editDataSourceFailMsg',
```

Comment:
editDataSourceFailMsg => Unable to edit the Data Source?

Reply:
If you look at the method fetchDataSourceDetailsByID() .. It is used to fetch the data source by ID .. If it fails, we show that error. So "Find" is appropriate.

Reply:
ok, however the code is using same id for different content, `dataSourcesManagement.editDataSource.editDataSourceFailMsg` this identifier will be used for localization, so if the content is different, consider use different id?

Reply:
Done